lib/access.js

```diff
@@ -1,30 +1,127 @@
+  return subcommands[sub].call(this, uri, params, cb)
+}
 
-  var options = {
+subcommands.public = function (uri, params, cb) {
+  return setAccess.call(this, 'public', uri, params, cb)
+}
+subcommands.private = function (uri, params, cb) {
+  return setAccess.call(this, 'private', uri, params, cb)
+}
+
+function setAccess (access, uri, params, cb) {
+  return this.request(apiUri(uri, 'package', params.package, 'access'), {
     method: 'POST',
-    body: JSON.stringify(body),
+    auth: params.auth,
+    body: JSON.stringify({ access: access })
+  }, cb)
+}
+
+subcommands.grant = function (uri, params, cb) {
+  var reqUri = apiUri(uri, 'team', params.scope, params.team, 'package')
+  return this.request(reqUri, {
+    method: 'PUT',
+    auth: params.auth,
+    body: JSON.stringify({
+      permissions: params.permissions,
+      package: params.package
+    })
+  }, cb)
+}
+
+subcommands.revoke = function (uri, params, cb) {
+  var reqUri = apiUri(uri, 'team', params.scope, params.team, 'package')
+  return this.request(reqUri, {
+    method: 'DELETE',
+    auth: params.auth,
+    body: JSON.stringify({
+      package: params.package
+    })
+  }, cb)
+}
+
+subcommands['ls-packages'] = function (uri, params, cb) {
+  var uriParams = '?format=cli'
+  var reqUri
+  if (params.team) {
+    reqUri = apiUri(uri, 'team', params.scope, params.team, 'package')
+    return this.request(reqUri + uriParams, {
+      method: 'GET',
+      auth: params.auth
+    }, cb)
+  } else if (params.scope) {
+    // TODO - support listing packages for orgs (just hit 'org' instead)
+    // NOTE - May need to just do a failover here to get it to work.
+    reqUri = apiUri(uri, 'user', params.scope, 'package')
+    return this.request(reqUri + uriParams, {
+      method: 'GET',
+      auth: params.auth
+    }, cb)
+  }
+}
+
+subcommands['ls-collaborators'] = function (uri, params, cb) {
+  var uriParams = '?format=cli'
+  var reqUri = apiUri(uri, 'package', params.package, 'collaborators')
+  return this.request(reqUri + uriParams, {
+    method: 'GET',
     auth: params.auth
+  }, cb)
+}
+
+// TODO - punted to v2
+// subcommands.edit = function (uri, params, cb) {
+//
+// }
+
+function apiUri (registryUri) {
+  var path = Array.prototype.slice.call(arguments, 1)
+    .map(encodeURIComponent)
+    .join('/')
+  return url.resolve(registryUri, '-/' + path)
+}
+
+function accessAssertions (subcommand, uri, params, cb) {
+  assert(subcommands.hasOwnProperty(subcommand),
+         'access subcommand must be one of ' +
+         Object.keys(subcommands).join(', '))
+  typeChecks({
+    'uri': [uri, 'string'],
+    'params': [params, 'object'],
+    'params.auth': [params.auth, 'object'],
+    'callback': [cb, 'function']
+  })
+  if (~[
+    'public', 'private', 'grant', 'revoke', 'ls-collaborators'
+  ].indexOf(subcommand)) {
+    typeChecks({ 'params.package': [params.package, 'string']})
+  }
+  if (~['grant', 'revoke', 'ls-collaborators'].indexOf(subcommand)) {
```

Comment:
`~` and `[].indexOf` interact in kind of a weird way (not that it's incorrect, it's just kind of strange taking the twos complement and calling that good because it just happens to evaluate to 0 when the element isn't found), so I think I'd be more comfortable with a more explicit `=== -1` here. (It's still your call, though.)